src/pymoca/tree.py

```diff
@@ -377,18 +388,38 @@ def build_instance_tree(orig_class: Union[ast.Class, ast.InstanceClass], modific
             # Fix component references to be one level deeper. E.g. applying a
             # modification "a.x = 3.0" on a symbol "a", will mean we pass
             # along a modification "x = 3.0" to the symbol's class instance.
+            # We should only do this if we are not (in)directly inheriting from a builtin.
+            inheriting_from_builtin = extends_builtin(c)
+            sym_mod = ast.ClassModification()
             for arg in sym_arguments:
                 if arg.value.component.indices:
                     raise Exception("Subscripting modifiers is not allowed.")
 
-                arg.value.component = arg.value.component.child[0]
+                if inheriting_from_builtin:
+                    for el_arg in arg.value.modifications:
+                        if not isinstance(el_arg, ast.ClassModification):
```

Comment:
Flip if/else blocks for clarity?

Reply:
Although I agree that checking for "not <something>" is a bit odd, I'd rather keep it the same as the equivalent check and block a couple lines up.